File: ext/hello_world/hello_world.c
#include <stdio.h>
#include <ruby.h>

static VALUE hello(VALUE klass)
{
  return rb_str_new2("Hello World from C...!");
}

void Init_hello_world(void) {
	VALUE helloWorld = rb_define_class("HelloWorld", rb_cObject);
  	rb_define_singleton_method(helloWorld, "hello", hello, 0);
}

File: ext/java/hello_world/HelloWorld.java
import org.jruby.Ruby;
import org.jruby.RubyClass;
import org.jruby.RubyObject;
import org.jruby.anno.JRubyClass;
import org.jruby.anno.JRubyMethod;

@JRubyClass(name="HelloWorld")
public class HelloWorld extends RubyObject {
	public HelloWorld(Ruby ruby, RubyClass rubyClass) {
        super(ruby, rubyClass);
    }

	@JRubyMethod(meta = true, rest = true)
    public static String hello() {
        return "Hello World from JAVA...!";
    }
}
